daemon/restart.go

```diff
@@ -39,8 +39,10 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		return err
 	}
 
-	if err := daemon.containerStart(container); err != nil {
-		return err
+	if !container.HostConfig.AutoRemove {
```

Comment:
This is limited by current design of restart policy, if we run `docker stop` on one container, it will stop immediately no matter what its restart policy is.  According to our previous discussion, `--rm` should mean "delete the container when the container stops and won't restart any more",  so if it stops, the container is removed, there's no need to start it again.

That's why I'm saying:
"4. `docker restart` a `--rm` container will behave like a `docker stop`, container will be stopped and removed."